components/brave_news/browser/brave_news_pref_manager.cc

```diff
@@ -0,0 +1,377 @@
+    for (const auto& channel : subscriptions) {
+      channel_ids.push_back(channel);
+    }
+  }
+
+  std::vector<std::string> other_channel_ids;
+  for (const auto& [locale, subscriptions] : other.channels_) {
+    for (const auto& channel : subscriptions) {
+      other_channel_ids.push_back(channel);
+    }
+  }
+  base::flat_set<std::string> channels_set(std::move(channel_ids));
+  base::flat_set<std::string> other_channels_set(std::move(other_channel_ids));
+
+  base::ranges::set_symmetric_difference(channels_set, other_channels_set,
+                                         std::back_inserter(result.changed));
+  return result;
+}
+
+// static
+void BraveNewsPrefManager::RegisterProfilePrefs(PrefRegistrySimple* registry) {
+  registry->RegisterBooleanPref(prefs::kShouldShowToolbarButton, true);
+  registry->RegisterBooleanPref(prefs::kNewTabPageShowToday,
+                                IsUserInDefaultEnabledLocale());
+  registry->RegisterBooleanPref(prefs::kBraveNewsOptedIn, false);
+  registry->RegisterDictionaryPref(prefs::kBraveNewsSources);
+  registry->RegisterDictionaryPref(prefs::kBraveNewsChannels);
+  registry->RegisterDictionaryPref(prefs::kBraveNewsDirectFeeds);
+  registry->RegisterBooleanPref(prefs::kBraveNewsOpenArticlesInNewTab, true);
+
+  p3a::NewsMetrics::RegisterProfilePrefs(registry);
+}
+
+BraveNewsPrefManager::BraveNewsPrefManager(PrefService& prefs) : prefs_(prefs) {
+  pref_change_registrar_.Init(&*prefs_);
+  pref_change_registrar_.Add(
+      prefs::kBraveNewsChannels,
+      base::BindRepeating(&BraveNewsPrefManager::NotifyChannelsChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kBraveNewsDirectFeeds,
+      base::BindRepeating(&BraveNewsPrefManager::NotifyPublishersChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kBraveNewsSources,
+      base::BindRepeating(&BraveNewsPrefManager::NotifyPublishersChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kBraveNewsOptedIn,
+      base::BindRepeating(&BraveNewsPrefManager::NotifyConfigChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kNewTabPageShowToday,
+      base::BindRepeating(&BraveNewsPrefManager::NotifyConfigChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      prefs::kBraveNewsOpenArticlesInNewTab,
+      base::BindRepeating(&BraveNewsPrefManager::NotifyConfigChanged,
+                          base::Unretained(this)));
+}
+
+BraveNewsPrefManager::~BraveNewsPrefManager() = default;
+
+void BraveNewsPrefManager::AddObserver(PrefObserver* observer) {
+  observers_.AddObserver(observer);
+}
+
+void BraveNewsPrefManager::RemoveObserver(PrefObserver* observer) {
+  observers_.RemoveObserver(observer);
+}
+
+bool BraveNewsPrefManager::IsEnabled() {
+  return GetIsEnabled(&*prefs_);
```

Comment:
this kind of thing should be a helper method in common so you can access the prefs without a dependency on all of  brave_news

Reply:
also why is being exposed as a method on `BraveNewsPrefManager` when it's already exposed as `brave_news::GetIsEnabled`?